src/IK/src/GravityTask.cpp

```diff
@@ -0,0 +1,208 @@
+{
+    constexpr auto errorPrefix = "[GravityTask::initialize]";
+
+    m_description = "GravityTask";
+
+    if (m_kinDyn == nullptr || !m_kinDyn->isValid())
+    {
+        log()->error("{} [{}] KinDynComputations object is not valid.", errorPrefix, m_description);
+
+        return false;
+    }
+
+    if (m_kinDyn->getFrameVelocityRepresentation()
+        != iDynTree::FrameVelocityRepresentation::MIXED_REPRESENTATION)
+    {
+        log()->error("{} [{}] task supports only quantities expressed in MIXED "
+                     "representation. Please provide a KinDynComputations with Frame velocity "
+                     "representation set to MIXED_REPRESENTATION.",
+                     errorPrefix,
+                     m_description);
+        return false;
+    }
+
+    auto ptr = paramHandler.lock();
+    if (ptr == nullptr)
+    {
+        log()->error("{} [{}] The parameter handler is not valid.", errorPrefix, m_description);
+        return false;
+    }
+
+    std::string robotVelocityVariableName;
+    if (!ptr->getParameter("robot_velocity_variable_name", m_robotVelocityVariable.name))
+    {
+        log()->error("{} [{}] while retrieving the robot velocity variable.",
+                     errorPrefix,
+                     m_description);
+        return false;
+    }
+
+    // set the gains for the controllers
+    if (!ptr->getParameter("kp", m_kp))
+    {
+        log()->error("{} [{}] to get the proportional linear gain.", errorPrefix, m_description);
+        return false;
+    }
+
+    std::string targetFrameName;
+    if (!ptr->getParameter("target_frame_name", targetFrameName))
+    {
+        log()->error("{} [{}] to get the end effector frame name.", errorPrefix, m_description);
+        return false;
+    }
+
+    m_targetFrameIndex = m_kinDyn->getFrameIndex(targetFrameName);
+
+    if (m_targetFrameIndex == iDynTree::FRAME_INVALID_INDEX)
+    {
+        log()->error("{} [{}] The specified target name ({}) does not seem to exist.",
+                     errorPrefix,
+                     m_description,
+                     targetFrameName);
+        return false;
+    }
+
+    m_isInitialized = true;
+
+    return true;
+}
+
+bool GravityTask::update()
+{
+    using namespace iDynTree;
+
+    m_isValid = false;
+
+    auto targetRotation = toEigen(m_kinDyn->getWorldTransform(m_targetFrameIndex).getRotation());
+
+    Eigen::Vector3d desiredZDirectionAbsolute = targetRotation * m_desiredZDirectionBody;
+    Eigen::Vector3d feedforwardAbsolute = targetRotation * m_feedForwardBody;
+
+    if (!m_kinDyn->getFrameFreeFloatingJacobian(m_targetFrameIndex, m_jacobian))
+    {
+        log()->error("[GravityTask::update] Unable to get the frame jacobian.");
+        return m_isValid;
+    }
+
+    toEigen(this->subA(m_robotVelocityVariable)) = m_Am * m_jacobian.bottomRows<3>();
+    m_b = -m_kp * m_bm * desiredZDirectionAbsolute + feedforwardAbsolute.topRows<2>();
```

Comment:
Taken from https://eigen.tuxfamily.org/dox/TopicWritingEfficientProductExpression.html

```suggestion
    m_b = feedforwardAbsolute.topRows<2>();
    m_b.noalias() -= m_kp * m_bm * desiredZDirectionAbsolute;
```